src/main/java/com/artipie/Credentials.java

```diff
@@ -89,19 +99,42 @@ public FromStorageYaml(final Storage storage, final Key key) {
         @Override
         public CompletionStage<List<String>> users() {
             return this.yaml().thenApply(
-                yaml -> yaml.yamlMapping("credentials").keys()
+                yaml -> yaml.yamlMapping(FromStorageYaml.CREDENTIALS).keys()
                     .stream().map(node -> node.asScalar().value()).collect(Collectors.toList())
             );
         }
 
         @Override
         public CompletionStage<Void> add(final String username, final String pswd) {
-            throw new IllegalArgumentException("Not implemented yet");
+            return this.yaml().thenApply(
+                yaml -> {
+                    YamlMappingBuilder result = FromStorageYaml.removeUserRecord(username, yaml);
+                    result = result.add(
+                        username,
+                        Yaml.createYamlMappingBuilder()
+                            .add("pass", String.format("plain:%s", pswd))
+                            .build()
+                    );
+                    return FromStorageYaml.buildCredentials(result);
+                }
+            ).thenCompose(
+                updated -> this.storage.save(
+                    this.key, new Content.From(updated.getBytes(StandardCharsets.UTF_8))
+                )
+            );
```

Comment:
@olenagerasimova this code snippet is repeated in `remove` method and it also proceeded with `FromStorageYaml.buildCredentials`. I'd suggest to extract method `save(YamlMapping)` that will build credentials YAML and save it to storage